pkg/scheduler/scheduler.go

```diff
@@ -183,6 +195,12 @@ func (s *Scheduler) schedule(ctx context.Context) wait.SpeedSignal {
 	log := ctrl.LoggerFrom(ctx).WithValues("schedulingCycle", s.schedulingCycle)
 	ctx = ctrl.LoggerInto(ctx, log)
 
+	var penalties *utilmaps.SyncMap[utilqueue.LocalQueueReference, corev1.ResourceList]
```

Comment:
instead of penalties let's initialize there the list of snapshotOptions, and only add the new param inside the FG.